modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -417,7 +424,11 @@ private IgniteBiTuple<Serializable, Boolean> handleWriteIntentSwitchCommand(
 
         UUID txId = cmd.txId();
 
-        txFinisher.markFinished(txId, cmd.commit(), cmd.commitTimestamp(), null);
+        if (!enabledColocationFeature) {
```

Comment:
That's a bit weird. We've agreed to introduce CommandHandlers both for better incapsulation and in order to reduce the rick of missing updates.
In case of WISC processing `PartitionListener#processCommand` still uses common method
```
        } else if (command instanceof WriteIntentSwitchCommand) {
            result = handleWriteIntentSwitchCommand((WriteIntentSwitchCommand) command, commandIndex, commandTerm);
```
thus second goal is not achieved.

Reply:
Agreed to add a comment with warning for now.